hs_core/management/commands/check_bag.py

```diff
@@ -0,0 +1,187 @@
+            dest='generate_metadata',  # value is options['generate_metadata']
+            help='force generation of metadata and bag'
+        )
+
+        parser.add_argument(
+            '--generate_bag',
+            action='store_true',  # True for presence, False for absence
+            dest='generate_bag',  # value is options['generate_bag']
+            help='force generation of metadata and bag'
+        )
+
+    def handle(self, *args, **options):
+
+        if len(options['resource_ids']) > 0:  # an array of resource short_id to check.
+            for rid in options['resource_ids']:
+                try:
+                    resource = BaseResource.objects.get(short_id=rid)
+                    istorage = resource.get_irods_storage()
+
+                    scimeta_path = os.path.join(resource.root_path, 'data',
+                                                'resourcemetadata.xml')
+                    if istorage.exists(scimeta_path):
+                        print("found {}".format(scimeta_path))
+                    else:
+                        print("{} NOT FOUND".format(scimeta_path))
+
+                    resmap_path = os.path.join(resource.root_path, 'data',
+                                               'resourcemap.xml')
+                    if istorage.exists(resmap_path):
+                        print("found {}".format(resmap_path))
+                    else:
+                        print("{} NOT FOUND".format(resmap_path))
+
+                    if istorage.exists(resource.bag_path):
+                        print("found bag {}".format(resource.bag_path))
+                    else:
+                        print("bag {} NOT FOUND".format(resource.bag_path))
+
+                    dirty = istorage.getAVU(resource.root_path, 'metadata_dirty')
+                    print("metadata_dirty is {}".format(str(dirty)))
+
+                    modified = istorage.getAVU(resource.root_path, 'bag_modified')
+                    print("bag_modified is {}".format(str(modified)))
+
+                    if options['generate']:  # generate usable bag
+
+                        create_bag_files(resource)
+                        print("metadata generated for {} from Django".format(rid))
+                        istorage.setAVU(resource.root_path, 'metadata_dirty', 'false')
+                        print("metadata_dirty set to false for {}".format(rid))
+
+                        create_bag_by_irods(rid)
+                        print("bag generated for {} from iRODs".format(rid))
+                        istorage.setAVU(resource.root_path, 'bag_modified', 'false')
+                        print("bag_modified set to false for {}".format(rid))
+
+                    elif options['generate_metadata']:
+
+                        create_bag_files(resource)
+                        print("metadata generated for {} from Django".format(rid))
+                        istorage.setAVU(resource.root_path, 'metadata_dirty', 'false')
```

Comment:
I wrote this so that it would work on develop and on www (for obvious reasons). Develop does not currently contain `AbstractResource.setAVU`. Will fix when release-candidate-1.10.0 is merged back into develop.